File: nestjs/src/modules/workspace/service/workspace-user/workspace-user.service.ts
import {InjectRepository} from "@nestjs/typeorm";
import {WorkspaceUser} from "../../../../database/entity/entity";
import {Repository} from "typeorm";
import {CreateWorkspaceUserDto} from "../../dto/createWorkspaceUserDto";
import {Injectable} from "@nestjs/common";

@Injectable()
export class WorkspaceUserService {
    constructor(
        @InjectRepository(WorkspaceUser)
        private workspaceUserRepository: Repository<WorkspaceUser>
    ) {}

    async createWorkspaceUser(createWorkspaceUserDto: CreateWorkspaceUserDto): Promise<WorkspaceUser> {
        const workspaceUser = new WorkspaceUser();
        workspaceUser.userId = createWorkspaceUserDto.userId;
        workspaceUser.nickname = createWorkspaceUserDto.nickname;
        return this.workspaceUserRepository.save(workspaceUser);
    }
}

File: go-psql/service/workspace_user_service.go
package service

import (
	"go-psql/database/repository"
	"go-psql/dto"

	"github.com/google/uuid"
)

type WorkspaceUserService struct {
	workspaceUserRepository repository.WorkspaceUserRepository
}

//factory

func NewWorkspaceUserService(workspaceUserRepository repository.WorkspaceUserRepository) WorkspaceUserService {
	return WorkspaceUserService{workspaceUserRepository: workspaceUserRepository}
}

//methods

func (wus *WorkspaceUserService) GetUser(id string) *dto.WorkspaceUser {
	return wus.workspaceUserRepository.Read(id)
}

func (wus *WorkspaceUserService) CreateUser(mongo_id, nickname string) error {
	var user dto.WorkspaceUser
	user.Id = uuid.New()
	user.UserId = mongo_id
	user.Nickname = nickname
	return wus.workspaceUserRepository.Create(user)
}
